resources/manifests/psp-restricted.yaml

```diff
@@ -0,0 +1,77 @@
+apiVersion: policy/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: restricted
+  annotations:
+    seccomp.security.alpha.kubernetes.io/allowedProfileNames: 'docker/default'
+    seccomp.security.alpha.kubernetes.io/defaultProfileName:  'docker/default'
+spec:
+  privileged: false
+  # Required to prevent escalations to root.
+  allowPrivilegeEscalation: false
+  # This is redundant with non-root + disallow privilege escalation,
+  # but we can provide it for defense in depth.
+  requiredDropCapabilities:
+  - KILL
+  - MKNOD
+  - SETUID
+  - SETGID
+  # Allow core volume types.
+  volumes:
+  - 'configMap'
+  - 'emptyDir'
+  - 'projected'
+  - 'secret'
+  - 'downwardAPI'
+  # Assume that persistentVolumes set up by the cluster admin are safe to use.
+  - 'persistentVolumeClaim'
+  hostNetwork: false
+  hostIPC: false
+  hostPID: false
+  runAsUser:
+    # Require the container to run without root privileges.
+    rule: 'MustRunAs'
+    ranges:
+    - min: 10000
+      max: 99999
+  seLinux:
+    rule: 'MustRunAs'
+    seLinuxOptions:
+      level: 's0:c11,c5'
```

Comment:
Where does this value comes from?

From what I remember from [slide 28-29](https://docs.google.com/presentation/d/1oBT3JGPFMDUC7CWvXf3SMEgUZopMJBAz6jizpB_vybk/edit#slide=id.g38b9e7c71a_0_5), the "category part (`c11.c5`) is for isolating containers from each others. So, if the PSP enforces all pods to have the same category, that would not isolate them? (but my understanding of SELinux is quite limited)

The other `seLinuxOptions` parameters (`user`, `role`, `type`) could be interesting too but they depends on the SELinux configuration of the underlying Linux distribution. On Flatcar Linux, I don't think it is enabled anyway.

-----

![image](https://user-images.githubusercontent.com/27575/46738581-dd082e00-cc9e-11e8-94c3-8dc5e1391bd5.png)
![image](https://user-images.githubusercontent.com/27575/46738399-65d29a00-cc9e-11e8-86b2-ec6e499cb82e.png)

Reply:
I think it would be fine to remove SELinux restrictions from our PSP for this first iteration.

Reply:
> I think it would be fine to remove SELinux restrictions from our PSP for this first iteration.

I second that :)

Reply:
done

Reply:
I learned that `seLinux` is a mandatory field, so we have to keep

```
diff --git a/resources/manifests/psp-restricted.yaml b/resources/manifests/psp-restricted.yaml
index 3b4dbdf..3976f3a 100644
--- a/resources/manifests/psp-restricted.yaml
+++ b/resources/manifests/psp-restricted.yaml
@@ -28,6 +28,8 @@ spec:
   hostNetwork: false
   hostIPC: false
   hostPID: false
+  seLinux:
+    rule: 'RunAsAny'
   runAsUser:
     # Require the container to run without root privileges.
     rule: 'MustRunAs'
```

Reply:
I learned that `seLinux` is a mandatory field, so we have to keep

```
diff --git a/resources/manifests/psp-restricted.yaml b/resources/manifests/psp-restricted.yaml
index 3b4dbdf..3976f3a 100644
--- a/resources/manifests/psp-restricted.yaml
+++ b/resources/manifests/psp-restricted.yaml
@@ -28,6 +28,8 @@ spec:
   hostNetwork: false
   hostIPC: false
   hostPID: false
+  seLinux:
+    rule: 'RunAsAny'
   runAsUser:
     # Require the container to run without root privileges.
     rule: 'MustRunAs'
```

Reply:
I learned that `seLinux` is a mandatory field, so we have to keep

```
diff --git a/resources/manifests/psp-restricted.yaml b/resources/manifests/psp-restricted.yaml
index 3b4dbdf..3976f3a 100644
--- a/resources/manifests/psp-restricted.yaml
+++ b/resources/manifests/psp-restricted.yaml
@@ -28,6 +28,8 @@ spec:
   hostNetwork: false
   hostIPC: false
   hostPID: false
+  seLinux:
+    rule: 'RunAsAny'
   runAsUser:
     # Require the container to run without root privileges.
     rule: 'MustRunAs'
```

Reply:
I learned that `seLinux` is a mandatory field, so we have to keep

```
diff --git a/resources/manifests/psp-restricted.yaml b/resources/manifests/psp-restricted.yaml
index 3b4dbdf..3976f3a 100644
--- a/resources/manifests/psp-restricted.yaml
+++ b/resources/manifests/psp-restricted.yaml
@@ -28,6 +28,8 @@ spec:
   hostNetwork: false
   hostIPC: false
   hostPID: false
+  seLinux:
+    rule: 'RunAsAny'
   runAsUser:
     # Require the container to run without root privileges.
     rule: 'MustRunAs'
```

Reply:
I learned that `seLinux` is a mandatory field, so we have to keep

```
diff --git a/resources/manifests/psp-restricted.yaml b/resources/manifests/psp-restricted.yaml
index 3b4dbdf..3976f3a 100644
--- a/resources/manifests/psp-restricted.yaml
+++ b/resources/manifests/psp-restricted.yaml
@@ -28,6 +28,8 @@ spec:
   hostNetwork: false
   hostIPC: false
   hostPID: false
+  seLinux:
+    rule: 'RunAsAny'
   runAsUser:
     # Require the container to run without root privileges.
     rule: 'MustRunAs'
```